public/js/reducers/sources.js

```diff
@@ -81,13 +79,19 @@ function update(state = initialState, action) {
   return state;
 }
 
-function getFilenameFromUrl(sourceUrl) {
+function _updateSource(source) {
+  const pathname = getPathnameFromUrl(source.url);
+  const filename = pathname.substring(pathname.lastIndexOf("/") + 1);
+  return Object.assign({}, source, { pathname, filename });
+}
+
+function getPathnameFromUrl(sourceUrl) {
   if (!sourceUrl) {
     return "";
   }
 
   const url = new URL(sourceUrl);
```

Comment:
If it's a selector and used from a component, I think it's ok to use `URL` and assume it's a browser environment. We don't test components in node, we'll run them in the browser, from what I understand